crates/router/src/db/user_role.rs

```diff
@@ -459,18 +460,22 @@ impl UserRoleInterface for MockDb {
         &self,
         user_id: &str,
         merchant_id: &str,
-    ) -> CustomResult<bool, errors::StorageError> {
+    ) -> CustomResult<storage::UserRole, errors::StorageError> {
         let mut user_roles = self.user_roles.lock().await;
-        let user_role_index = user_roles
+
+        match user_roles
             .iter()
-            .position(|user_role| {
-                user_role.user_id == user_id && user_role.merchant_id == merchant_id
-            })
-            .ok_or(errors::StorageError::ValueNotFound(format!(
-                "No user available for user_id = {user_id}"
-            )))?;
-        user_roles.remove(user_role_index);
-        Ok(true)
+            .position(|role| role.user_id == user_id && role.merchant_id == merchant_id)
+        {
+            Some(index) => {
+                let deleted_user_role = user_roles.remove(index);
+                Ok(deleted_user_role)
```

Comment:
No need of assignment here as well.
```suggestion
Ok(user_roles.remove(index))
```